src/redisraft.c

```diff
@@ -651,6 +651,11 @@ static void handleRedisCommandAppend(RedisRaftCtx *rr,
         return;
     }
 
+    if (!(cmd_flags & CMD_SPEC_SCRIPTS) && validateAllCommandsACL(ctx, cmds) != REDISMODULE_OK) {
+        RedisModule_ReplyWithError(ctx, "NOPERM this user doesn't have proper permissions");
```

Comment:
`RM_ACLCheckCommandPermissions` returns only OK/ERROR, which doesn't reflect the real root cause of the error, so we cannot reply in the same reply msg as redis. We maybe can consider changing the api to return also the acl return val - `ACL_DENIED_CMD\ACL_DENIED_KEY\ACL_DENIED_CHANNEL`, it should be easy because `ACLCheckAllUserCommandPerm()`  already returns this value. @sjpotter , WDYT?

Reply:
the better response would be to either return the key (as the underlying call does) along with the error or just pass in a RedisModuleString ** and if not null, it generates the proper error message.  I thought that was a bit overkill for now.  In redis itself, it had a default error message when it can't parse.